opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -1896,7 +1896,9 @@ namespace {
                     use_free_gas
                 };
 
-                this->updateGuideRateModel(guide_rate_model, report_step);
+                // A NONE target written to .UNRST may indicate no GUIDERAT (i.e., during history)
+                if (! (guide_rate_model.target()==GuideRateModel::Target::NONE))
+                    this->updateGuideRateModel(guide_rate_model, report_step);
```

Comment:
Maybe we could push this test to the enclosing `if()` condition?  For instance, something along the lines of
```C++
const auto target = GuideRateModel::TargetFromRestart(header.guide_rate_nominated_phase);
if ((target != GuideRateModel::Target::None) &&
    GuideRateModel::rst_valid(...))
{
    // as before
}
```
That way, we won't even have to do any other checking if target is `NONE`.

Reply:
Good point, will update.